spec/shared_context.rb

```diff
@@ -121,3 +116,30 @@ def integration_context_debugging
     end
   end
 end
+
+def system_context
```

Comment:
The name of this method is misleading... This is not only the system context, it also loads a lot of common vars, secrets, clients...

Please split it in two different methods.

Also, you left some of the *shared client load and mock* dangling in the `before(:each)`. Please bring it here before the method split.